Tasks/VsTestPlatformToolInstallerV1/nugetfeedinstaller.ts

```diff
@@ -103,9 +126,52 @@ export class NugetFeedInstaller {
         }
 
         vstestPlatformInstalledLocation = await new NugetDownloadHelper(this.consolidatedCiData)
-            .attemptPackageDownload(packageSource, testPlatformVersion);
+            .attemptPackageDownload(packageSource, testPlatformVersion, nugetConfigFilePath);
 
         // Set the vstest platform tool location for the vstest task to consume
         helpers.setVsTestToolLocation(vstestPlatformInstalledLocation);
     }
+
+    // Utility function that writes the feed url along with username and password if provided into the specified nuget config file
+    private prepareNugetConfigFile(packageSource: string, configFilePath: string, username: string, password: string) : string {
+        const feedUrl = tl.getInput(constants.customFeed);
+        const feedId = uuid.v1();
+
+        tl.debug(`Writing package source details to temp config file ${configFilePath}`);
+
+        try {
+            // Write the skeleton nuget config contents to the config file
+            fs.writeFileSync(configFilePath, constants.emptyNugetConfig, { encoding: 'utf-8' });
+        } catch (error) {
+            this.consolidatedCiData.failureReason = 'configFileWriteFailed';
+            throw new Error(tl.loc('ConfigFileWriteFailed', configFilePath, error));
+        }
+
+        if (!helpers.isNullEmptyOrUndefined(password) && helpers.isNullEmptyOrUndefined(username)) {
+            username = 'vstestPlatformToolInstaller';
+        }
+
+        const nugetTool = tl.tool(path.join(__dirname, 'nuget.exe'));
+
+        nugetTool.arg(constants.sources).arg(constants.add).arg(constants.noninteractive)
+            .arg(constants.name).arg(feedId).arg(constants.source).arg(feedUrl)
+            .argIf(password, constants.usernameParam).argIf(password, username)
+            .argIf(password, constants.passwordParam).argIf(password, password)
+            .argIf(configFilePath, constants.configFile).argIf(configFilePath, configFilePath);
+
+        this.consolidatedCiData.prepareConfigFileStartTime = perf();
```

Comment:
curious why do you need start time and end time? you need execution time

Reply:
yeah. better to store the execution time.

Reply:
here there's the problem of cluttering with too many local variables. Today in the queries we are any how performing a subtraction. Not entirely sure what would be best here